erp/views.py

```diff
@@ -193,104 +193,127 @@ def _update_search_pager(pager, commune):
     return pager
 
 
-def _clean_search_params(params, *args):
-    return ("" if params.get(arg, "") == "None" else params.get(arg, "") for arg in args)
+def _cleaned_search_params_as_dict(get_parameters):
+    allow_list = ("where", "what", "lat", "lon", "code", "search_type", "postcode")
+    return {
+        k: "" if get_parameters.get(k, "") == "None" else get_parameters.get(k, "")
+        for k, v in get_parameters.items()
+        if k in allow_list
+    }
 
 
 def _parse_location_or_404(lat, lon):
-    if not lat or not lon:
+    if not all([lat, lon]):
         return None
     try:
         return geo.parse_location((lat, lon))
     except RuntimeError as err:
         raise Http404(err)
 
 
-def _get_search_qs_and_context(commune_slug, lat, lon, what, where):
-    location = _parse_location_or_404(lat, lon)
-    qs = Erp.objects.select_related("accessibilite", "activite", "commune_ext").published().search_what(what)
-    commune = None
-    if commune_slug:
-        commune = get_object_or_404(Commune, slug=commune_slug)
-        qs = qs.filter(commune_ext=commune)
-        where = str(commune)
-    elif location:
-        qs = qs.nearest(location)
-    elif where and where not in ("France entière", translate("France entière")):
-        location = geocoder.autocomplete(where)
-        if location:
-            lat, lon = (location.y, location.x)  # update current searched lat/lon
-            qs = qs.nearest(location)
-        else:
-            qs = qs.search_commune(where)
-
-    return qs, {
-        "commune": commune,
-        "lat": lat,
-        "lon": lon,
-        "what": what,
-        "where": where,
-        "commune_json": commune.toTemplateJson() if commune else None,
+def _filter_erp_by_location(queryset, **kwargs):
+    search_type = kwargs.get("search_type")
+
+    # TODO setting for this ?
+    if search_type == "municipality":
+        return queryset.filter(commune_ext=kwargs.get("municipality"))
+
+    if search_type == settings.ADRESSE_DATA_GOUV_SEARCH_TYPE_CITY:
+        postcode = kwargs.get("postcode")
+        return queryset.filter(commune_ext__code_postaux__contains=[postcode])
```

Comment:
Do this in 2 steps, retrieve commune from post code and then use the commune name in the ERP